src/libstore/filetransfer.cc

```
@@ -580,7 +580,12 @@ struct curlFileTransfer : public FileTransfer
         #endif
 
         #if __linux__
-        unshareFilesystem();
+        try {
+            tryUnshareFilesystem();
```

Shouldn't the try/catch be moved inside `tryUnshareFilesystem()` since it claims to be best-effort?

---

Maybe it should never throw an exception? Are there any errors that should not be ignored?

---

I am not sure, but I am generally uncomfortable with ignoring errors without a whitelist.